lib/commands/command-plugin-test.bash

```diff
@@ -14,7 +14,8 @@ plugin_test_command() {
   local plugin_url="$2"
   shift 2
 
-  local plugin_gitref="master"
+  plugin_remote_default_branch=$(git ls-remote --symref $plugin_url HEAD | awk '{ sub(/refs\/heads\//, ""); print $2; exit }')
```

Comment:
```suggestion
  plugin_remote_default_branch=$(git ls-remote --symref "$plugin_url" HEAD | awk '{ sub(/refs\/heads\//, ""); print $2; exit }')
```

Reply:
Thank you so much for your review!

Would it be OK if I just move the check to be executed before?
```diff
diff --git a/lib/commands/command-plugin-test.bash b/lib/commands/command-plugin-test.bash
index 16f5c1d..0cb2659 100644
--- a/lib/commands/command-plugin-test.bash
+++ b/lib/commands/command-plugin-test.bash
@@ -14,6 +14,19 @@ plugin_test_command() {
   local plugin_url="$2"
   shift 2
 
+  local exit_code
+  local TEST_DIR
+
+  fail_test() {
+    printf "FAILED: %s\n" "$1"
+    rm -rf "$TEST_DIR"
+    exit 1
+  }
+
+  if [ -z "$plugin_name" ] || [ -z "$plugin_url" ]; then
+    fail_test "please provide a plugin name and url"
+  fi
+
   plugin_remote_default_branch=$(git ls-remote --symref $plugin_url HEAD | awk '{ sub(/refs\/heads\//, ""); print $2; exit }')
   local plugin_gitref=${3:-${plugin_remote_default_branch}}
   local tool_version
@@ -50,19 +63,6 @@ plugin_test_command() {
     set -- "${SHELL:-sh}" -c "$1"
   fi
 
-  local exit_code
-  local TEST_DIR
-
-  fail_test() {
-    printf "FAILED: %s\n" "$1"
-    rm -rf "$TEST_DIR"
-    exit 1
-  }
-
-  if [ -z "$plugin_name" ] || [ -z "$plugin_url" ]; then
-    fail_test "please provide a plugin name and url"
-  fi
-
   TEST_DIR=$(mktemp -dt asdf.XXXX)
   cp -R "$(asdf_dir)/bin" "$TEST_DIR"
   cp -R "$(asdf_dir)/lib" "$TEST_DIR"
(END)
```

Reply:
If it were me, I would move the `plugin_gitref` assignment to right after the `case` statement because I think the argument parsing logic is more readable when it is at the top of the file - it would also be consistent with the other `command-*` files, I believe.

Reply:
I see your point. Nevertheless, I'm finding it challenging to maintain the suggested ordering elegantly. The current setup with the `plugin_gitref` assignment first setting the default value aligns with the logic of the case statement, which overwrites it when a value for `--asdf-plugin-gitref` is provided. Any insights on how to approach it?

Reply:
How about checking if it `plugin_gitref` is empty with `[ -z`? That way, if no value or an empty value is supplied, we fallback to calculating it on our own

Reply:
Makes sense to me. I commited some changes. Let me know if this version fits :)

Reply:
That works - the new changes look good to me